docs/src/conversion.md

```diff
@@ -48,6 +48,105 @@ The exceptions are as follows.
 
 ## Explicit GAP-to-Julia and Julia-to-GAP Conversions
 
+The following rules hold for explicit conversions.
+
+1. Julia types control the conversions.
+
+   - For conversions from Julia to GAP,
+     there is at most one possibility on the GAP side,
+     and the type of the given Julia object decides which code is used.
+
+   - For conversions from GAP to Julia, several Julia types can be possible
+     for the result, for example a GAP integer can be converted to several
+     Julia integer types, or a GAP string can be converted to a Julia `String`
+     or `Tuple`.
+     Usually one wants to specify the target type,
+     and then this type decides which code is used for the conversion.
+     If one does not specify the target type, a default type will be chosen.
+
+2. Subobjects, recursive conversions
+
+   - GAP lists and records can have subobjects,
+     the same holds for various Julia objects such as vectors, matrices,
+     tuples, and dictionaries.
+     One may or may not want to convert the subobjects recursively,
+     this is controlled by the `recursive` keyword argument of the functions
+     `GAP.gap_to_julia` and `GapObj`, which can be set to `true` or `false`.
+
+   - For Julia-to-GAP conversion, the default is non-recursive conversion.
+
+   - For GAP-to-Julia conversion, the default is recursive conversion.
+
+   - For Julia-to-GAP conversion, recursion stops at subobjects of type
+     `GapObj`.
+
+   - For GAP-to-Julia conversion, recursion stops at subobjects that do
+     not have the type `GapObj`.
```

Comment:
Maybe I am misunderstanding something, or maybe I am too picky, but: What about converting a GAP list like `[1,2,3]` to `Vector{Int16}` ? Technically the entries of the GAP list are not `GapObj`, they are `Int`... Yet we do "recurse" to convert those integers (good!), as we see here:
```gap-repl
julia> l = GapObj([1,2,3])
GAP: [ 1, 2, 3 ]

julia> Vector{Int16}(l)
3-element Vector{Int16}:
 1
 2
 3

julia> Vector{Float32}(l)
3-element Vector{Float32}:
 1.0
 2.0
 3.0
```

Reply:
Maybe referring to `GAP.Obj` instead would do the job?

Reply:
Yes, we can do that.

On the other hand, I would say that the given example does not fit:
The prescribed type `Vector{Int16}` forces a conversion of the list entries to `Int16` independent of the decision about recursion.